src/main/resources/static/js/MainWindow.js

```diff
@@ -1,769 +1,743 @@
+        .find("select option:selected")
+        .val();
+      if (currentInput == null || currentInput == "") {
+        $.notify(
+          { message: "Please leave no datastream empty." },
+          {
+            allow_dismiss: true,
+            type: "info",
+            placement: { from: "top", align: "left" },
+            animate: {
+              enter: "animated fadeInDown",
+              exit: "animated fadeOutUp"
+            }
+          }
+        );
+        stop = true;
+        return false;
+      }
+      obj["dsID"] = JSON.parse(
+        $(this)
+          .find("select option:selected")
+          .attr("data-value")
+      ).frostId;
+      $(this)
+        .find("tbody tr")
+        .each(function() {
+          currentInput = $(this)
+            .find("td:eq(1) input")
+            .val();
+          if (currentInput == null || currentInput == "") {
+            $.notify(
+              {
+                message:
+                  "Please specify the column where the observations of each datastream can be found (no empty field allowed) before saving the configuration."
+              },
+              {
+                allow_dismiss: true,
+                type: "info",
+                placement: { from: "top", align: "left" },
+                animate: {
+                  enter: "animated fadeInDown",
+                  exit: "animated fadeOutUp"
+                }
+              }
+            );
+            stop = true;
+            return false;
+          }
+
+          parsed = parseInt(currentInput, 10);
+
+          if (!(currentInput == parsed) || parsed < 0) {
+            $.notify(
+              {
+                message:
+                  "Please specify the column where the observations of each datastream can be found (must be a non-negative number) before saving the configuration."
+              },
+              {
+                allow_dismiss: true,
+                type: "info",
+                placement: { from: "top", align: "left" },
+                animate: {
+                  enter: "animated fadeInDown",
+                  exit: "animated fadeOutUp"
+                }
+              }
+            );
+            stop = true;
+            return false;
+          }
+          obs.push(parsed);
+        });
+      if (stop == true) {
+        return false;
+      }
+
+      obj["observations"] = obs;
+      obj["multiStream"] = obs.length > 1;
+      streams.push(obj);
+    });
+
+  if (stop == true) {
+    return false;
+  }
+
+  var filetype = null;
+  if ($("input[name=source]:eq(0)").is(":checked")) {
+    filetype = $("#file").val();
+  } else if ($("input[name=source]:eq(1)").is(":checked")) {
+    filetype = $("#sourceinput").val();
+  }
+  if (!filetype || filetype == "") {
+    addToLog("No File specified.");
+  } else {
+    filetype = filetype
+      .split(".")
+      .pop()
+      .toUpperCase();
+    if (filetype == "XLSX" || filetype == "XLS") {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected '===' and instead saw '=='.](https://app.codacy.com/gh/FraunhoferIOSB/ChillImport/pullRequest?prid=4898788)